File: pfe-java/app/models/SocialNetwork.java
package models;

import controllers.URL;
import play.libs.F;
import play.libs.Scala;
import play.libs.ws.WS;
import play.libs.ws.WSClient;
import play.libs.ws.WSResponse;
import play.mvc.Http;

public class SocialNetwork {

    WSClient ws = WS.client();

    static final String SHARING_ENDPOINT = "http://www.mocky.io/v2/539c5a907650aa6202515c00";

    public F.Promise<WSResponse> share(String content, String token) {
        return ws.url(SHARING_ENDPOINT)
                .setQueryParameter("access_token", token)
                .setContentType(Http.MimeTypes.FORM)
                .post(URL.encode(Scala.varargs(URL.param("content", content))));
    }

    public static final SocialNetwork SocialNetwork = new SocialNetwork();

}

File: pfe-scala/service/src/main/scala/models/SocialNetwork.scala
package models

import play.api.libs.ws.{WSClient, WSResponse}
import scala.concurrent.Future

class SocialNetwork(ws: WSClient) {

  val sharingEndpoint = "http://www.mocky.io/v2/539c5a907650aa6202515c00"

  def share(content: String, token: String): Future[WSResponse] =
    ws.url(sharingEndpoint)
      .withQueryString("access_token" -> token)
      .post(Map("content" -> Seq(content)))

}
